Jenkinsfile

```
@@ -38,8 +38,12 @@ pipeline {
                         }
                     }
                     steps {
-                        echo 'Running Cargo check...'
-                        sh 'cargo check --all --all-targets --features sdl_controller,json,saveload'
+                        // Make sure static_assertion fails
+                        echo 'Confirming static assertion...'
+                        sh 'cargo check --all --all-targets || exit 0'
```

should we go `&& exit 1` because we want it to fail on success?

---

That command will fail because no backend feature was specified. So we want it to fail to succeed :)

---

ya i mean:

```bash
fail && exit 1 || exit 0 # succeeds on failure, fails on pass
```

---

@azriel91 is right, we definitely need to fail in case this command succeeds, but we must also not exit if it fails! That would skip the command below.

I'd suggest the following: 

```suggestion
                        sh 'cargo check --all --all-targets && exit 1 || true'
```

**Note:** This is technically no if-then-else; `&&` / `||` have left-associativity, thus in `A && B || C`, `C` will run if either `A` or `B` failed. This is only okay here because `B` will just terminate the whole command. Usually, you'll want to use an actual if-then-else block here.

---

Applied as suggested.